src/global/tracking/TrackPropagation_factory.cc

```diff
@@ -58,15 +58,15 @@ void eicrecon::TrackPropagation_factory::Process(const std::shared_ptr<const JEv
 
     for(auto traj: trajectories) {
         edm4eic::MutableTrackSegment this_propagated_track;
-        for(size_t isurf = 0; auto surf: m_target_surface_list) {
+        for(size_t isurf = 0; isurf < m_target_surface_list.size(); isurf++) {
+            auto surf = m_target_surface_list[isurf];
```

Comment:
Does this work?
```suggestion
        for(auto& surf : m_target_surface_list) {
```
i.e. can we then get `surface` and `system` from `surf`?

Reply:
If it is not easy to do that, and it would delay this PR substantially, then we can just file an issue and proceed with the fix as is, though.

Reply:
Unfortunately I do not think so - or at least I don't know how. We currently get the system id like so:

```
auto BEMC_system_id = m_geoSvc->detector()->constant<uint32_t>("ECalBarrel_ID");
```

so basically it gets the constant by name from `dd4hep::Detector*` - I am at least not familiar enough to know if it is possible to get directly from the surface? Or rather store it on the surface object somehow

Reply:
It seems we already store them in the surface ID, e.g.
```cpp
    BEMC_prop_surface1->assignGeometryId(Acts::GeometryIdentifier().setExtra(BEMC_system_id).setLayer(1));
    BEMC_prop_surface2->assignGeometryId(Acts::GeometryIdentifier().setExtra(BEMC_system_id).setLayer(2));
```

Reply:
The geometry ID `extra` field is the 8 lower bits of the unsigned long int, and we store detector ID in it for all 'regular' surfaces defined in the geometry. So we can use them here as well, and they will be propagated to the 8 lower bits of the surface field in the stored track points. (Layer is stored in a different set of bits in the geometry ID. Ref: https://acts.readthedocs.io/en/latest/core/geometry/geometry_id.html)